### DISCUSSION TITLE
Unique ID for Content Part

### DISCUSSION THREAD

**Discussion by jameswoodley** [OrchardCMS/OrchardCore#12112] [Q&A]
Is it possible to add an ID to a content part? We have a content part that allows you to add a collection of items, but it's just Title and Subtitle. While we can currently get at them via index, if that index changes (user reorders the items) we lose the reference. I'd love to be able to assign an ID automatically to them that Orchard stores and is output in the GraphQL.

I know I could add a textfield for the user to add an ID, but this is very manual and error prone.

Any ideas gratefully received!

**Comment by jptissot** [comment]
Would need to know more about this content part you have ? Is it built in code or in the Admin UI ?

**Comment by jameswoodley** [comment]
It's in the admin ui

**Comment by ns8482e** [comment]
You could add alternate template or you could use wrapper shape to add id

I normally add wrapper using placement

**Reply by jameswoodley** [reply]
Can you help me here, I'm not sure I get what you mean by wrapper shape?
For context, these are the items I'd like unique ids on 
![image](https://user-images.githubusercontent.com/52610563/181786485-6fa550d5-5f37-492d-b768-c05c39f18446.png)

**Reply by ns8482e** [reply]
If you are using bag part and content items then each content item you add has content item id
